checkstyle-tester/diff.groovy

```diff
@@ -628,36 +625,39 @@ def getProjectsStatistic(diffDir) {
     return projectsStatistic
 }
 
-def runMavenExecution(srcDir, excludes, checkstyleConfig,
-                      checkstyleVersion, extraMvnRegressionOptions) {
-    println "Running 'mvn clean' on $srcDir ..."
-    def mvnClean = "mvn -e --no-transfer-progress --batch-mode clean"
-    executeCmd(mvnClean)
-    println "Running Checkstyle on $srcDir ... with excludes {$excludes}"
-    def mvnSite = "mvn -e --no-transfer-progress --batch-mode site " +
-        "-Dcheckstyle.config.location=$checkstyleConfig -Dcheckstyle.excludes=$excludes"
-    if (checkstyleVersion) {
-        mvnSite = mvnSite + " -Dcheckstyle.version=$checkstyleVersion"
+def runCliExecution(allJar, mainClassOptions) {
+    def saveLoc = new File(mainClassOptions.saveDir)
+    if (!saveLoc.exists()) {
+        saveLoc.mkdirs()
     }
-    if (extraMvnRegressionOptions) {
-        mvnSite = mvnSite + " "
 
-        if (!extraMvnRegressionOptions.startsWith("-")) {
-            mvnSite = mvnSite + "-"
+    println "Running Checkstyle CLI on ${mainClassOptions.srcDir} ... with excludes {${mainClassOptions.excludes}}"
+    def cliCommand = "java -jar $allJar -c ${mainClassOptions.checkstyleConfig} " +
```

Comment:
I can't help but feel that StringBuilder would be faster/easier to read/use less memory, especially with a bunch of excludes.

Reply:
You want me to switch?

Reply:
Keep in mind performance is not an issue here as it is one time execution. Better to keep code easy to read. I don't mind any approach

Reply:
It's not a big deal, but we certainly run this more than once (twice for each project). I was thinking more for readability since `+` at the end of the line bugs me. I will leave it up to @rnveach

Reply:
we need to disable trailing "+" rule.
I always thought about groovy as Java script abilities, to write in pure Java but as script. All scripts grows and grow up in applications, so being plain java is good for them

Reply:
It is not a rule, this is required by the interpreter

Reply:
@romani Please see the message I posted for you at https://github.com/checkstyle/checkstyle/pull/12725#discussion_r1098907617 . This is groovy itself, and not a style rule.

Reply:
sad, but ok, lets follow interpreter.